Switch Burgers, ScalarWave, and NewtonianEuler to new evolution scheme

nilsdeppe opened the pull request:
## Proposed changes

I've tested that LTS works correctly with ScalarWave, and also that issues we saw in the past with HLL and shocks leaving the grid are now fixed.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsdeppe committed:
Switch Burgers to new evolution scheme

nilsdeppe committed:
Switch NewtonianEuler to new evolution scheme

nilsdeppe committed:
Switch scalar wave to new evolution scheme

kidder commented on tests/InputFiles/Burgers/Step.yaml:
```diff
@@ -21,18 +21,20 @@ DomainCreator:
 
 SpatialDiscretization:
+  BoundaryCorrection:
+    Hll:
```
just checking if it was intentional to switch to HLL

nilsdeppe replied:
Yea, it's what people should be using, I think :)

nilsdeppe commented:
rebased on develop to deal with conflicts. Thanks for the review :)

nilsdeppe commented:
rebased again on develop

nilsdeppe commented:
I've rebased again to resolve conflicts. Cheers!

fmahebert commented on src/Evolution/Systems/Burgers/System.hpp:
```diff
@@ -25,6 +27,10 @@ struct System {
 
+  using boundary_conditions_base = BoundaryConditions::BoundaryCondition;
+
+  using boundary_correction = BoundaryCorrections::BoundaryCorrection;
```
For a future consistency improvement, should this type alias be renamed `boundary_correction_base` so that it better matches the `boundary_conditions_base` above?

fmahebert commented on tests/InputFiles/Burgers/Step.yaml:
```diff
@@ -30,18 +30,20 @@ DomainCreator:
 
-NumericalFlux:
-  LocalLaxFriedrichs:
 
```
I think you've left two blank lines... if so please delete one of them

fmahebert commented on tests/InputFiles/ScalarWave/PlaneWave1D.yaml:
```diff
@@ -51,8 +55,6 @@ SpatialDiscretization:
 
-NumericalFlux:
-  UpwindPenalty:
 
```
same comment

fmahebert commented on tests/Unit/PointwiseFunctions/AnalyticSolutions/WaveEquation/CMakeLists.txt:
```diff
@@ -4,10 +4,15 @@
+# it over to the ComputeTimeDerivate and ApplyBoundaryCorrections actions
+# is too time consuming at the moment (Mar. 25, 2021). We should:
+# - switch it over later
+# - switch to a pypp test that we use as a regression test
```
Can you clarify if this list is an "and" or an "or"? Perhaps by adding either ", and" or ", or" at the end of the first bullet.

fmahebert reviewed (commented):
Please squash directly

nilsdeppe committed:
Rename boundary_correction to boundary_correction_base

This is to be consistent with boundary_condition_base

nilsdeppe commented:
I've rebased and:
- added a new commit that renames `boundary_correction` to `boundary_correction_base`
- added fixups to do the rename in the executables
- removed a few other extra type aliases that are no longer needed (char speeds and magnitude)

Thanks for the reviews!

kidder reviewed (commented):
you have my permission to squash

fmahebert reviewed (commented):
LGTM, please squash

nilsdeppe committed:
Add missing includes to Test_ScalarWaveGr.cpp

nilsdeppe commented:
Rebased and squashed. I also added a commit that adds some of the missing includes to `Test_ScalarWaveGr.cpp` (I didn't go through the entire file in detail checking each line)

Thanks for the reviews :)
